distributed/tests/test_scheduler.py

```diff
@@ -3274,6 +3274,26 @@ async def test__to_dict(c, s, a, b):
     assert d["tasks"][futs[0].key]
 
 
+@gen_cluster(client=True, nthreads=[])
+async def test_TaskState__to_dict(c, s):
+    """tasks that are listed as dependencies of other tasks are dumped as a short repr
+    and always appear in full under Scheduler.tasks
+    """
+    x = c.submit(inc, 1, key="x")
+    y = c.submit(inc, x, key="y")
+    z = c.submit(inc, 2, key="z")
+    while len(s.tasks) < 3:
+        await asyncio.sleep(0.01)
+
+    tasks = s._to_dict()["tasks"]
+
+    assert isinstance(tasks["x"], dict)
+    assert isinstance(tasks["y"], dict)
+    assert isinstance(tasks["z"], dict)
+    assert tasks["x"]["dependents"] == ["<TaskState 'y' waiting>"]
```

Comment:
I'm trying to figure out why this couldn't race. At some point, both of these will transition to `"processing"` or `"memory"`. It passes locally just fine for me, BTW, but makes me nervous.

Why not
```python
    assert tasks["x"]["dependents"] == [repr(cluster.scheduler.tasks["y"])]
    assert tasks["y"]["dependencies"] == [repr(cluster.scheduler.tasks["x"])]
```
?

Reply:
There are no workers, therefore the tasks are never transitioned. X is in state `no-worker` indicating it cannot be assigned. y is in `waiting` since it waits for x to complete first.